ts/environment.ts

```diff
@@ -3,59 +3,118 @@ import { hasTag, nixSource } from "./internal/utils.ts";
 import { Check } from "./check.ts";
 import { Executable } from "./executable.ts";
 import {
-  Interpolatable,
-  nixList,
+  NixStrLitInterpolatable,
   NixExpression,
+  nixList,
   nixRaw,
   nixStrLit,
 } from "./nix.ts";
 
+/**
+ * `Environment`s define what files and tools are available to `Executables`,
+ * `Check`s and development shells.
+ *
+ * For example they can contain compilers and developer tools that you want to
+ * use on a project, but they may also define other things like environment
+ * variables that need to be set.
+ *
+ * You can enter an `Environment` with `garn enter`.
+ */
 export type Environment = {
   tag: "environment";
   nixExpression: NixExpression;
+  /**
+   * Creates a new environment based on this one that includes the specified nix packages.
+   */
```

Comment:
As mentioned in #108 deno doc does not support rendering these yet unfortunately. However I think the fact that they show up on LSP hover is useful still.